### PR TITLE
ccs-qcd: Change compile option for aarch64

### PR DESCRIPTION
**Pull request by t-nojiri**
Fixed option string when compiling with gcc compiler.
The description of  allowed value of variant was changed.
Change to use join_path() instead of the path separator "/" to eliminate OS dependency.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ccs-qcd/package.py [hunk 1]
```diff
@@ -33,24 +19,33 @@ class CcsQcd(MakefilePackage):
     version('master', branch='master')
     version('1.2.1', commit='d7c6b6923f35a824e997ba8db5bd12dc20dda45c')
 
-    variant('class', values=int, default=1,
+    variant('class', default=1, values=('1', '2', '3', '4', '5', '6'),
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ccs-qcd/package.py]: @alalazo @scheibelp @becker33 I believe we discovered a bug recently where values either can't be ints or can't be strings?

Comment (alalazo) [reply var/spack/repos/builtin/packages/ccs-qcd/package.py]: I'm not sure of the bug you are referring to, might have missed that in the stream of notifications :thinking:  Here it seems to me we are also deleting the error message in the deleted function, is it intentional?

Comment (t-nojiri) [reply var/spack/repos/builtin/packages/ccs-qcd/package.py]: class_validator was intentionally deleted, but I will modify it to use it.

File: var/spack/repos/builtin/packages/ccs-qcd/package.py [hunk 2]
```diff
@@ -33,24 +34,33 @@ class CcsQcd(MakefilePackage):
     version('master', branch='master')
     version('1.2.1', commit='d7c6b6923f35a824e997ba8db5bd12dc20dda45c')
 
-    variant('class', values=int, default=1,
+    variant('class', default=1, values=class_validator,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ccs-qcd/package.py:37]: Doesn't `values` have to be a list, not a function?

File: var/spack/repos/builtin/packages/ccs-qcd/package.py (hunk 1 above)
Comment (t-nojiri) [reply var/spack/repos/builtin/packages/ccs-qcd/package.py]: When values ​​is a list, the intended error message is not output.

Comment (t-nojiri) [reply var/spack/repos/builtin/packages/ccs-qcd/package.py]: $ spack info ccs-qcd class=7
==> Error: unrecognized arguments: class=7

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/ccs-qcd/package.py]: I think that's because the default is an int but the values are strings.

Comment (t-nojiri) [reply var/spack/repos/builtin/packages/ccs-qcd/package.py]: The intended error message is what is output by class_validator.
$ spack install ccs-qcd %gcc@8.3.1 class=7
==> Error: class: Choose one of the following:
1  - 8x8x8x32 (default MPI config: 1x1x1)
2  - 32x32x32x32 (default MPI config: 4x4x4)
3  - 64x64x64x32 (default MPI config: 8x8x8)
4  - 160x160x160x160 (default MPI config: 20x20x20)
5  - 256x256x256x256 (default MPI config: 32x32x32)
6  - 192x192x192x192 (default MPI config: 24x24x24)

**Review by becker33 — APPROVED** [review]

**Pull request merged**